apps/src/code-studio/pd/workshop_dashboard/components/survey_results/single_choice_responses.jsx

```diff
@@ -0,0 +1,70 @@
+import React, {PropTypes} from 'react';
+import _ from 'lodash';
+import {Panel} from 'react-bootstrap';
+
+export default class SingleChoiceResponses extends React.Component {
+  static propTypes = {
+    question: PropTypes.string.isRequired,
+    answers: PropTypes.object.isRequired,
+    answerType: PropTypes.string.isRequired,
+    possibleAnswers: PropTypes.array.isRequired,
+    otherAnswers: PropTypes.array
+  };
+
+  render() {
+    let totalAnswers = Object.values(this.props.answers).reduce((sum, x) => sum + x, 0);
+    return (
+      <Panel>
+        {this.props.question}
+        <table>
+          <tbody>
+          {
+            this.props.possibleAnswers.map((possibleAnswer, i) => {
+              let answerIndex;
+
+              if (this.props.answerType === 'selectValue') {
+                answerIndex = i + 1;
+              } else {
+                answerIndex = possibleAnswer;
```

Comment:
I'm confused by this `answerIndex`. How is the index the same as the `possibleAnswer` for non-value questions? Won't that be text in all other cases?

Reply:
For selectValue questions, the value is not the text of the response but a numeric value in some range. The hash contains a map of the numeric values to counts, not texts to counts.

(I'll add this comment)